platforms/gke/base/use-cases/federated-learning/examples/cross-device/README.md

```diff
@@ -0,0 +1,164 @@
+microservices. These components do the following:
+
+- Collector: this job runs periodically to query active tasks and encrypted
+  gradients. This information determines when aggregation starts
+- Task-assignment: this front-end service distributes training tasks to devices
+- Task-management: this job manages tasks
+- Task-scheduler: this job either runs periodically or is triggered by specific
+  events
+- Task-builder: this job builds tasks to be sent to clients
+
+Two of the FCP components have to be run in a confidential space virtual machine
+that ensures the memory is encrypted during use. These components are:
+
+- Aggregator: this job reads device gradients and calculates aggregated result
+  with Differential Privacy
+- Model updater: this job listens to events and publishes results so that device
+  can download updated models
+
+In addition to these components, the architecture also deploys:
+
+- A GCS bucket to store the consolidated model and gradients
+- A pubsub that enables communication between the different microservices
+- A spanner table that register the tasks to be sent and the history of the
+  training tasks
+
+## Understand the repository structure
+
+The cross-device example has the following directories and files.
+
+In the `platforms/gke/base/use-cases/federated-learning/examples/cross-device`
+directory:
+
+- `assets`: contains documentation static assets.
+- `deploy.sh`: convenience script to deploy an instance of the cross-device
+  example on the reference architecture.
+- `teardown.sh`: convenience script to destroy the cross-device example
+  instance.
+- `setup-environment.sh`: contains common shell variables and functions for the
+  cross-device example.
+- `README.md`: this document.
+
+In the
+`platforms/gke/base/use-cases/federated-learning/terraform/example_cross_device`:
+
+- Terraform descriptors to deploy Google Cloud resources.
+- Cross-device configuration file template.
+- GKE and Cloud Service Mesh descriptors to deploy and expose the cross-device
+  workloads.
+
+## Deploy the reference architecture
+
+This example builds on top of the infrastructure that the
+[Federated Learning reference architecture](/platforms/gke/base/use-cases/federated-learning/README.md)
+provides, and follows the best practices that the reference architecture
+establishes.
+
+After you deploy the reference architecture instances, continue following this
+document.
+
+## Build the images needed
+
+In order to completely deployed this infrastructure, you will need to build and
+push on a registry the images to run the workloads. These images can be found in
+the
+[On-Device Personalization Federated Compute Server](https://github.com/privacysandbox/odp-federatedcompute).
+
+1. Clone and update the repository
+
+   ```bash
+   git clone https://github.com/privacysandbox/odp-federatedcompute
+   git submodule update --init --recursive
+   ```
+
+1. Create a worker-pool large enough to build the images
```

Comment:
Terraform this. Worth doing this in it's own terraservice (`cross_device_cloud_build`?)